presto-main/src/test/java/com/facebook/presto/memory/TestNodeMemoryConfig.java

```diff
@@ -14,33 +14,28 @@
 package com.facebook.presto.memory;
 
 import com.google.common.collect.ImmutableMap;
+import io.airlift.configuration.testing.ConfigAssertions;
 import io.airlift.units.DataSize;
 import org.testng.annotations.Test;
 
 import java.util.Map;
 
+import static com.facebook.presto.memory.LocalMemoryManager.validateHeapHeadroom;
+import static com.facebook.presto.memory.NodeMemoryConfig.AVAILABLE_HEAP_MEMORY;
 import static io.airlift.configuration.testing.ConfigAssertions.assertFullMapping;
 import static io.airlift.units.DataSize.Unit.BYTE;
 import static io.airlift.units.DataSize.Unit.GIGABYTE;
-import static org.testng.Assert.fail;
 
 public class TestNodeMemoryConfig
 {
     @Test
     public void testDefaults()
     {
-        // This can't use assertRecordedDefaults because the default value is dependent on the current max heap size, which varies based on the current size of the survivor space.
-        NodeMemoryConfig config = new NodeMemoryConfig();
-        long maxMemory = Runtime.getRuntime().maxMemory();
-        for (int i = 0; i < 1_000; i++) {
-            DataSize expectedQueryMaxMemory = new DataSize(maxMemory * 0.1, BYTE);
-            DataSize expectedQueryMaxTotalMemory = new DataSize(maxMemory, BYTE);
-            if (expectedQueryMaxMemory.equals(config.getMaxQueryMemoryPerNode()) && expectedQueryMaxTotalMemory.equals(config.getMaxQueryTotalMemoryPerNode())) {
-                return;
-            }
-        }
-        // We can't make this 100% deterministic, since we don't know when the survivor space will change sizes, but assume that something is broken if we got the wrong answer 1000 times
-        fail();
+        ConfigAssertions.assertRecordedDefaults(
+                ConfigAssertions.recordDefaults(NodeMemoryConfig.class)
```

Comment:
You can merge this line with the previous one. If you choose to put this on a separate line, you would need to indent the following lines.